google/cloud/bigtable/row_response.py

```diff
@@ -36,47 +37,270 @@ class RowResponse(Sequence["CellResponse"]):
     cells = row["family", "qualifier"]
     """
 
-    def __init__(self, key: row_key, cells: list[CellResponse]):
+    def __init__(
+        self,
+        key: row_key,
+        cells: list[CellResponse]
+        | dict[tuple[family_id, qualifier], list[dict[str, Any]]],
+    ):
+        """
+        Initializes a RowResponse object
+
+        RowResponse objects are not intended to be created by users.
+        They are returned by the Bigtable backend.
+        """
         self.row_key = key
-        self.cells: OrderedDict[
-            family_id, OrderedDict[qualifier, list[CellResponse]]
+        self._cells_map: dict[
+            family_id, dict[qualifier, list[CellResponse]]
```

Comment:
I think a better data structure would be: OrderedDict[family, list[Cell]) and use bsearch to find a qualifier in the cell list